src/scripts/effectEngine/effectEngineRunner.ts

```diff
@@ -0,0 +1,25 @@
+class effectEngineRunner {
+    public static counter: number = 0;
+
+    public static tick() {
+        this.counter = 0;
+        
+        for(let item in player.effectEngine){
+            if(item != 'equals'){
```

Comment:
What is this 'equals'?

Reply:
without it - an extra equals: NaN object is added. no clue why or where - but its there lol. this check removes that item. Could be something related to knockout but no clue.

Reply:
its because of [`Array.prototype.equals` here](https://github.com/Ishadijcks/pokeclicker/blob/develop/src/scripts/safari/SafariBody.ts#L350)

Reply:
I've applied a fix here https://github.com/Ishadijcks/pokeclicker/pull/285/commits/50214c7b95076b4cff03c48210670f37ff40575c
By changing the object from `[]` → `{}`,
But this does mean if you already have the `effectEngine` object attached to your save, you will need to apply `player.effectEngine = {}` manually